src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -1901,24 +1901,48 @@ private static bool GetDayOfNNY(ref DateTimeResult result, scoped ref DateTimeRa
 
             if (order == ORDER_MDY || order == ORDER_YMD)
             {
-                if (SetDateYMD(ref result, raw.year, n1, n2))
-                {
-                    result.flags |= ParseFlags.HaveDate;
-                    return true; // MD + Year
-                }
+                if (SetYMD(ref result, raw.year, n1, n2))
```

Comment:
and `SetDateYMD` -> `SetYMD`. Could you please explain.

Also DateTimeParse is probably perf critical, what is the timing comparison ?

Reply:
`result.flags |= ParseFlags.HaveDate;` is inside of the `SetYMD`/`SetYDM` functions. The renaming was not intentional, thanks for spotting it.
The performance cannot change much: I wrapped the existing code into another function so that it's easily seen which is setting MD and which DM (before only analysing the order of arguments that had similar names gave us this info) + I added a check for `GlobalizationMode.Hybrid` flag for Browser platform. It does not look like a change that changes performance.